internal/querycoordv2/meta/channel_dist_manager.go

```diff
@@ -345,13 +347,18 @@ func (m *ChannelDistManager) updateCollectionIndex() {
 func (m *ChannelDistManager) GetShardLeader(channelName string, replica *Replica) *DmChannel {
 	m.rwmutex.RLock()
 	defer m.rwmutex.RUnlock()
-
+	logger := log.With(zap.String("Scope", "Rolling"), zap.String("channelName", channelName), zap.Int64("replicaID", replica.GetID()))
 	channels := m.collectionIndex[replica.GetCollectionID()]
 
 	var candidates *DmChannel
-	for _, channel := range channels {
+	for chIdx, channel := range channels {
+		logger := logger.With(zap.Int("channelIdx", chIdx))
+		logger.Debug("process", zap.String("channel", channel.GetChannelName()),
+			zap.Int64("channelID", channel.Node), zap.Int64("channelVersion", channel.Version),
+			zap.Any("channelView", channel.View), zap.Bool("replicaContains", replica.Contains(channel.Node)))
 		if channel.GetChannelName() == channelName && replica.Contains(channel.Node) {
 			if candidates == nil {
+				logger.Debug("set candidate channel", zap.String("channel", channel.GetChannelName()))
```

Comment:
zap.String("channel", channel.GetChannelName() is redundant, and this logs is unnecessary, cause this code path is definitely executed